data/openstreetmaps/src/main/java/de/mm20/launcher2/openstreetmaps/OsmSerialization.kt

```diff
@@ -0,0 +1,21 @@
+package de.mm20.launcher2.openstreetmaps
+
+import de.mm20.launcher2.search.SavableSearchable
+import de.mm20.launcher2.search.SearchableDeserializer
+import de.mm20.launcher2.search.SearchableSerializer
+
+class OsmLocationSerializer : SearchableSerializer {
+    override fun serialize(searchable: SavableSearchable): String {
+        searchable as OsmLocation
+        TODO()
+    }
+
+    override val typePrefix: String
+        get() = "osmlocation"
+}
+
+class OsmLocationDeserializer : SearchableDeserializer {
+    override suspend fun deserialize(serialized: String): SavableSearchable? {
+        TODO()
```

Comment:
TODO

Reply:
I implemented this with network calls now. Otherwise, we may also just store ID, lat and lon and deserialize that (but information would be incomplete).

Anyway, we need to query OSM sooner or later anyway when retrieving deserialized data, since as you mentioned, we might display outdated information otherwise. The question is where we implement this call.

Reply:
Bad idea. The launcher relies on serializers completing within milliseconds because this potentially runs very frequently (every time anything changes in the Searchable database table, so every time an item is pinned, unpinned, hidden, or launcher). If it doesn't complete in time, the entire launcher is slowed down by this.

Here is an alternative approach that might work:
- store ID, name, lat, lon, and category in the database
- add `suspend fun`s to the Location interface to access information that is subject to change (i.e. `suspend fun getOpeningHours`)
- load the information lazily the first time it is requested by the UI, then cache them for the lifetime of the Location object